code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -13,23 +13,61 @@
 	var/list/model_type = null
 	/// Bitflags listing model compatibility. Used in the exosuit fabricator for creating sub-categories.
 	var/list/model_flags = NONE
-	// if true, is not stored in the robot to be ejected
-	// if model is reset
+
+	/// List of items to add with the module, if any
+	var/list/items_to_add = list()
+	/// List of items to remove with the module, if any
+	var/list/items_to_remove = list()
```

Comment:
```suggestion
	/// List of items to add with the module, if any
	var/list/items_to_add
	/// List of items to remove with the module, if any
	var/list/items_to_remove
```
These can be null by def since you're using **length()** and not **.len**